src/app.py

```diff
@@ -458,10 +460,125 @@ def notify():
+        # it MUST be a 401 error with message.
+        # That's how commons.auth_helper.getAuthorizationTokens() was designed
+        if isinstance(user_token, Response):
+            # We wrap the message in a json and send back to requester as 401 too
+            # The Response.data returns binary string, need to decode
+            unauthorized_error(user_token.get_data().decode())
+
+        # Also check if the parsed token is invalid or expired
+        # Set the second parameter as False to skip group check
+        user_info = auth_helper_instance.getUserInfo(user_token, False)
+
+        if isinstance(user_info, Response):
+            unauthorized_error(user_info.get_data().decode())
+
 ####################################################################################################
 ## Ingest API Endpoints
 ####################################################################################################
 
+"""
+For each element in a list of identifiers, return accessibility information appropriate
+for the user authorization of the Request.
+
+An HTTP 400 Response is returned for reasons described in the error message, such as
+not providing the list of identifiers.
+
+An HTTP 401 Response is returned when a token is presented that is not valid.
+
+An HTTP 500 is returned for unexpected errors
+
+Parameters
+----------
+request : flask.request
+    The flask http request object that containing the Authorization header
+    with a valid Globus groups token for checking group information. The
+    Request will have the Content-type header set to application/json. The
+    JSON body of the request will contain a JSON Array of strings with
+    UUID or HuBMAP-ID strings.
+
+Returns
+-------
+json
+    Valid JSON for a single JSON Object containing only JSON Objects, one per
+    entity evaluated.  This enclosing Object will have keys for each identifier
+    submitted with the request, whose value is a JSON Object containing
+    accessibility information for the entity. Each entity JSON Object will contain
+    "valid_id": true/false,  --true if the id resolves to a HuBMAP Dataset or Upload
+    ----------  below here only returned if valid_id == true
+    "access_allowed": true/false --true if the user is allowed to access the data for this entity
+    ----------  below here only returned if access_allowed == true
+    "hubmap_id": "<corresponding HuBMAP ID of the requested id>",
+    "uuid": "<uuid of Dataset or Upload>",
+    "entity_type": "<Dataset or Upload>",
+    "file_system_path": "<full absolute file system path to the Dataset or upload>"
+"""
+@app.route('/entities/accessible-data-directories', methods=['POST'])
+def get_accessible_data_directories():
+    dataset_helper = DatasetHelper()
+
+    # If not token is provided or an invalid token is provided, return a 401 error.
+    if request.headers.get('Authorization') is None:
+        unauthorized_error('A valid token must be provided.')
```

Comment:
As a result of the newly added gateway auth check, this check is no longer needed here.

Reply:
This is from me attempting to adapt to the style of this codebase, and mimic other endpoints in app.py.  I've removed this check.